spec.html

```diff
@@ -171,6 +172,18 @@ <h1>Web Scripting</h1>
     <p>Each Web browser and server that supports ECMAScript supplies its own host environment, completing the ECMAScript execution environment.</p>
   </emu-clause>
 
+  <emu-clause id="sec-hosts-and-implementations">
+    <h1>Hosts and Implementations</h1>
+    <p>To aid integrating ECMAScript into host environments, this specification defers the definition certain facilities, either in whole or in part, to a source outside of this specification. Editorially, this specification distinguishes the following kinds of deferrals.</p>
+    <p>An <dfn>implementation</dfn> is an external source that further defines facilities enumerated in Annex <emu-xref href="#sec-host-layering-points"></emu-xref> or those that are marked as implementation-defined or implementation-approximated. In informal use, an implementation refers to a concrete artefact, such as a particular web browser.</p>
+    <p>An <dfn>implementation-defined</dfn> facility is one that defers its definition to an external source without further qualification. This specification does not make any recommendations for particular behaviours, and conforming implementations are free to choose any behaviour within the constraints put forth by this specification.</p>
+    <p>An <dfn>implementation-approximated</dfn> facility is one that defers its definition to an external source while recommending an ideal behaviour. While conforming implementations are free to choose any behaviour within the constraints put forth by this specification, they are encouraged to strive to approximate the ideal. Mathematical operations, such as `Math.exp`, are implementation-approximated.</p>
+    <p>A <dfn>host</dfn> is an external source that further defines facilities enumerated in Annex <emu-xref href="#sec-host-layering-points"></emu-xref> but does not further define other implementation-defined or implementation-approximated facilities. In informal use, a host refers to the set of all implementations, such as the set of all web browsers, that interface with this specification in the same way via Annex <emu-xref href="#sec-host-layering-points"></emu-xref>. A host is often an external specification, such as WHATWG HTML. In other words, facilities that are host-defined are often further defined in external specifications.</p>
+    <p>A <dfn>host hook</dfn> is an abstract operation that is defined by an external source.</p>
```

Comment:
A host hook is any abstract operation that's not 100% defined in ecma262. The taxonomy is:

- A facility is a thing that ecma262 defines.
- Any facility that isn't 100% defined in ecma262 is called implementation-defined.
- There's an annex that lists a subset of implementation-defined facilities.
- A host-defined facility is any facility that's listed in that annex.
- A host hook is an implementation-defined abstract operation. All such AOs must be listed in that annex.
- All host-defined things are by definition implementation-defined, but should be referred to as host-defined to be specific.

The idea is HTML can leverage this taxonomy to say it is a host, since it doesn't further define things like Math.exp. Conforming implementations of HTML itself aren't free to redefine the host-defined parts, since while ecma262 alone allows it, those implementations must conform to both HTML and ecma262.